### PR TITLE
BaSP-MR-39: test trainer get post

### PR DESCRIPTION
**Pull request by Dannylez**
Seed and tests (get, getbyid and post) for trainers

### CODE REVIEW

File: src/tests/trainer.test.js [hunk 1]
```diff
@@ -1 +1,152 @@
...
+  });
+  test('Error 409 if repeated email', async () => {
+    const response = await request(app)
+      .post('/api/trainers')
+      .send(mockRepeatedEmail);
+    expect(response.status).toBe(409);
+    expect(response.body.message).toBe(
+      'Trainer with that Email already exists',
+    );
+    expect(response.error).toBeTruthy();
```
Comment (IgnacioGalceran) [inline src/tests/trainer.test.js:137]: here we should verify if we can access correctly to the response.body.error

File: src/tests/trainer.test.js [hunk 2]
```diff
@@ -1 +1,152 @@
...
+  });
+  test('Error 409 if repeated dni', async () => {
+    const response = await request(app)
+      .post('/api/trainers')
+      .send(mockRepeatedDni);
+    expect(response.status).toBe(409);
+    expect(response.body.message).toBe('Trainer with that DNI already exists');
+    expect(response.error).toBeTruthy();
```
Comment (IgnacioGalceran) [inline src/tests/trainer.test.js:145]: the same here, and in every reponse.error

File: src/tests/trainer.test.js [hunk 3]
```diff
@@ -1 +1,152 @@
...
+    expect(response.status).toBe(404);
+    expect(response.error).toBeTruthy();
+  });
+});
+
+describe('POST /api/trainers', () => {
+  test('Post correctly when the right data is sent', async () => {
+    const response = await request(app).post('/api/trainers').send(mockTrainer);
+    expect(response.status).toBe(201);
+    expect(response.body.data).toHaveProperty(
```
Comment (IgnacioGalceran) [inline src/tests/trainer.test.js:115]: here you can also verify if expect(response.body.data).toEqual(mockTrainer) or something like that?

File: src/tests/trainer.test.js (hunk 1 above)
Comment (Dannylez) [reply src/tests/trainer.test.js]: fixed

File: src/tests/trainer.test.js (hunk 3 above)
Comment (Dannylez) [reply src/tests/trainer.test.js]: fixed

File: src/tests/trainer.test.js (hunk 2 above)
Comment (Dannylez) [reply src/tests/trainer.test.js]: fixed

**Review by IgnacioGalceran — APPROVED** [review]
✈🚀

**Review by nailaghigo — COMMENTED** [review]
💯

**Pull request merged**